app.js

```diff
@@ -0,0 +1,108 @@
+var path = require('path');
+var fs = require('fs');
+var express = require('express');
+var Router = require('react-router');
+
+var indexStaticWatcher = require('./lib/index-static-watcher').create();
+var PORT = process.env.PORT || 8008;
+var PRODUCTION = (process.env.NODE_ENV === 'production');
+var DIST_DIR = path.join(__dirname, 'dist');
+var WATCH_DELAY = 300;
+
+var indexStatic;
+var router;
+var app = express();
+
+var startApp = function() {
+  app.listen(PORT, function() {
+    console.log('Listening on port', PORT);
+  });
+};
+
+var startDevApp = function() {
+  var gulp = require('gulp');
+  var gulpfile = require('./gulpfile');
+
+  require('./lib/developer-help')();
+
+  indexStaticWatcher.watch(WATCH_DELAY, function(newIndexStatic) {
+    updateIndexStatic(newIndexStatic);
+    console.log('Rebuilt server-side bundle.');
+  });
+
+  gulp.start('watch-webpack');
+  gulp.start('less');
+  gulp.start('copy-static-files');
+  gulp.start('watch-static-files');
+
+  gulp.watch(gulpfile.LESS_FILES, ['less']).on('change', function() {
+    console.log('Rebuilding LESS files.');
+  });
+
+  startApp();
+};
+
+var startProdApp = function() {
+  console.log([
+    'Production mode enabled. Note that "npm install" is assumed to',
+    'have recently been run with NODE_ENV="production". If this is not',
+    'the case, some or all static assets may be out of date.'
+  ].join('\n'));
```

Comment:
For more details on the rationale behind this, see [this comment](https://github.com/mozilla/teach.webmaker.org/pull/936/files#r30858286).